Assets/Editor/Tests/GZipStreamWrapperTest.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.IO;
+using System.Text;
+using Ionic.Zlib;
+using NUnit.Framework;
+using PatchKit.Unity.Utilities;
+
+public class GZipStreamWrapperTest
+{
+    private class MockStream : Stream
+    {
+        public override bool CanRead
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override bool CanSeek
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override bool CanWrite
+        {
+            get
+            {
+                return false;
+            }
+        }
+        public override long Length
+        {
+            get
+            {
+                return _internalStream.Length;
+            }
+        }
+
+        public override long Position
+        {
+            get
+            {
+                return _internalStream.Position;
+            }
+
+            set
+            {
+                _internalStream.Position = value;
+            }
+        }
+
+        private readonly int _packetSize;
+
+        private readonly MemoryStream _internalStream;
+
+        public MockStream(byte[] data, int packetSize)
+        {
+            _packetSize = packetSize;
+            _internalStream = new MemoryStream(data);
+        }
+
+        public override void Flush()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return _internalStream.Read(buffer, offset, System.Math.Min(count, _packetSize));
```

Comment:
You can make it random instead of `Min`, it would make this test more accurate.

Reply:
No, I don't think that would quite work.

Reply:
Hm okay.